### PR TITLE
[SIM_FLUTTER-107] [INTEGRATE] Implement Category In Withdrawal

### PR DESCRIPTION
**Pull request by IanUrriza10**
### Issue Links
- [backlog](https://framgiaph.backlog.com/view/SIM_FLUTTER-107)
- [slack](https://sun-philippine.slack.com/archives/C059L7XAXFB/p1689328795399319)

### Definition of done
- [x] Withdraw successfully with all categories
 
### Screenshots

https://github.com/framgia/sph-flutter/assets/114897466/fd137f27-f2da-4066-a0da-28423f8632e0

image

### CODE REVIEW

**Review by Dbulawan13 — DISMISSED** [review]
Approve PR #70 

Working:
- Able to implement category withdrawal
- Able to withdraw
- Current balance will be deducted in realtime
- Validations still work

https://github.com/framgia/sph-flutter/assets/86758474/37f02032-b663-443d-b190-d43dcd99a154

Testing Hours: 1
Comment: 0
Iteration: 0

**Comment by Dbulawan13** [comment]
Comment:

- [ ] Food/Drinks should be selected by default. 


Comment: 1

**Review by Dbulawan13 — DISMISSED** [review]

File: frontend/lib/src/components/transaction_component/transaction_component.dart [hunk 1]
```diff
@@ -69,21 +69,10 @@ class TransactionComponent extends StatelessWidget {
...
-      },
-      TransactionTypes.DEPT: {
-        'category': TransactionCategories.SAVINGS,
-        'success': 'deposited',
-      },
-      TransactionTypes.TRANSFER: {
-        'category': TransactionCategories.SENDER,
-        'success': 'transfered'
-      },
+    final Map<TransactionTypes, String> typeTexts = {
```
Comment (jstephend-sun) [inline frontend/lib/src/components/transaction_component/transaction_component.dart]:
```suggestion
    final Map<TransactionTypes, String> x = {
```
rename to a more descriptive variable

File: frontend/lib/src/components/transaction_component/transaction_component.dart [hunk 2]
```diff
@@ -69,21 +69,10 @@ class TransactionComponent extends StatelessWidget {
...
+      TransactionTypes.CREDIT: 'withdrawn',
+      TransactionTypes.DEPT: 'deposited',
+      TransactionTypes.TRANSFER: 'transfered',
```
Comment (jstephend-sun) [inline frontend/lib/src/components/transaction_component/transaction_component.dart]:
```suggestion
      TransactionTypes.TRANSFER: 'transferred',
```
fix spelling

**Review by jstephend-sun — CHANGES_REQUESTED** [review]
Requested Changes #70

review time: 5mins
comments: 2
iteration: 1st review

**Review by jstephend-sun — APPROVED** [review]
Approved #70

review time: 5mins
comments: 2
iteration: 1st review

**Pull request merged**